lib/client/policy_e2e_test.go

```diff
@@ -0,0 +1,206 @@
+			log.Println("Get policy list returns: ", policyList.Items)
+
+			// Get a policy with meta1.
+			outPolicy1, outError1 = c.Policies().Get(meta1)
+
+			// Assert they are equal and no errors.
+			Expect(policyList.Items[0].Spec).To(Equal(outPolicy1.Spec))
+			Expect(outError1).NotTo(HaveOccurred())
+
+			By("Delete, Get and assert error")
+
+			// Delete a policy with meta1.
+			outError1 = c.Policies().Delete(meta1)
+
+			// Get a policy with meta1.
+			_, outError = c.Policies().Get(meta1)
+
+			// Expect an error since the policy was deleted.
+			Expect(outError.Error()).To(Equal(errors.New("resource does not exist: Policy(name=policy1)").Error()))
+
+			// Delete the second policy with meta2.
+			outError1 = c.Policies().Delete(meta2)
+
+			By("Delete all the policies, Get policy list and expect empty policy list")
+
+			// Both policies are deleted in the calls above.
+			// Get the list of all the policys.
+			policyList, outError = c.Policies().List(api.PolicyMetadata{})
+			log.Println("Get policy list returns: ", policyList.Items)
+
+			// Create an empty policy list.
+			// Note: you can't use make([]api.Policy, 0) because it creates an empty underlying struct,
+			// whereas new([]api.Policy) just returns a pointer without creating an empty struct.
+			emptyPolicyList := new([]api.Policy)
+
+			// Expect returned policyList to contain empty policyList.
+			Expect(policyList.Items).To(Equal(*emptyPolicyList))
+
+		},
+
+		// Test 1: Pass two fully populated PolicySpecs and expect the series of operations to succeed.
+		Entry("Two fully populated PolicySpecs",
+			api.PolicyMetadata{Name: "policy1"},
+			api.PolicyMetadata{Name: "policy2"},
+			*createAPIPolicySpecObject("policy1", 99.999, "policy1-selector"),
+			*createAPIPolicySpecObject("policy2", 22.222, "policy2-selector"),
+		),
+
+		// Test 2: Pass one fully populated PolicySpec and another empty PolicySpec and expect the series of operations to succeed.
+		Entry("One fully populated PolicySpec and another empty PolicySpec",
+			api.PolicyMetadata{Name: "policy1"},
+			api.PolicyMetadata{Name: "policy2"},
+			*createAPIPolicySpecObject("policy1", 99.999, "policy1-selector"),
+			api.PolicySpec{},
+		),
+
+		// Test 3: Pass one partially populated PolicySpec and another fully populated PolicySpec and expect the series of operations to succeed.
+		Entry("One partially populated PolicySpec and another fully populated PolicySpec",
+			api.PolicyMetadata{Name: "policy1"},
+			api.PolicyMetadata{Name: "policy2"},
+			api.PolicySpec{
+				Selector: "policy1-selector",
+			},
+			*createAPIPolicySpecObject("policy2", 22.222, "policy2-selector"),
+		),
+	)
+})
+
+// createAPIPolicySpecObject takes policy configuration options (name, order, selector),
+// creates 2 fixed set of ingress and egress rules (one with IPv4 and one with IPv6),
+// and composes & returns an api.PolicySpec object.
+func createAPIPolicySpecObject(name string, order float64, selector string) *api.PolicySpec {
```

Comment:
Alternatively, you could define a set of Rules that you use in both the policy and profile tests.